test/built-ins/WeakRef/prototype/constructor.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2019 Leo Balter. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-properties-of-the-weak-ref-prototype-object
+description: WeakRef.prototype.constructor
+info: |
+  WeakRef.prototype.constructor
+
+  Normative Optional
+
+  The initial value of WeakRef.prototype.constructor is the intrinsic object %WeakRef%.
+
+  This property has the attributes { [[Writable]]: false, [[Enumerable]]: false, [[Configurable]]: true }.
+
+  This section is to be treated identically to the "Annex B" of ECMA-262, but to be written in-line with the main specification.
+includes: [propertyHelper.js]
+features: [WeakRef]
+---*/
+
+var actual = WeakRef.prototype.hasOwnProperty('constructor');
+
+// If implemented, it should conform to the spec text
```

Comment:
This test should be written in a way that engines which implement Annex B can reliably assert that the property exists. The previous directory structure provided a way to do this; we shouldn't regress just because normative optional is inline. Maybe `normative-optional` could be a feature flag.

Reply:
this flag would make the tests results rely more on how the metadata is interpreted and this needs some coordination with people who maintain test runners. Otherwise the js code would just straight forward force a feature.

I'd prefer to discuss this in a follow up as well, if you don't mind.